### PR TITLE
Start a fresh daemon for tests.

### PR DESCRIPTION
**Pull request by tomprince**
This means that running tests on a system with an existing MPD running doesn't mess it up.

It also means that we can eventually add files and things to it, to test more of the functionality of the client.

TODO:
- [x] Deal with daemon output.

### CODE REVIEW

**Comment by tomprince** [comment]
This is ready for review.

File: tests/helpers/daemon.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            db_file: base.join("db"),
+            music_directory: base.join("music"),
+            playlist_directory: base.join("playlists"),
+            config_path: base.join("config"),
+            sock_path: base.join("sock"),
+        }
+    }
+
+    fn config_text(&self) -> String {
+        format!(r#"
```
Comment (NoctuaNivalis) [inline tests/helpers/daemon.rs]: Perhaps move this string to a constant on top, this kind of breaks the code flow.

File: tests/helpers/daemon.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+pub struct Daemon {
+    // Saved here so it gets dropped when this does.
+    _temp_dir: TempDir,
+    config: MpdConfig,
+    process: Child,
+}
+
+impl Drop for Daemon {
```
Comment (NoctuaNivalis) [inline tests/helpers/daemon.rs]: Aren't trait implementations usually placed below the general impl?

File: tests/helpers/daemon.rs (hunk 1 above)
Comment (kstep) [inline tests/helpers/daemon.rs]: I don't see a problem here. Now you can see the generated text immediately in the place it is generated. If we move this string to some other place, we will have to look it up to understand the function output better.

File: tests/helpers/daemon.rs (hunk 2 above)
Comment (tomprince) [inline tests/helpers/daemon.rs]: I'm not sure if there is any particular conventions around this. I'm happy to move things around if desired.

File: tests/helpers/daemon.rs (hunk 1 above)
Comment (NoctuaNivalis) [inline tests/helpers/daemon.rs]: Just personal preference and autism.

File: tests/helpers/daemon.rs (hunk 2 above)
Comment (kstep) [inline tests/helpers/daemon.rs]: I don't have any strong preferences on this part. I'll leave it on the PR's author opinion.

**Comment by kstep** [comment]
Something's still wrong with nightly build, but it doesn't seem to be related to rust-mpd, so I'll merge it for now.

**Pull request merged**